databases/week2/task4.sql

```diff
@@ -0,0 +1,59 @@
+CREATE DATABASE smoothie;
+
+USE smoothie;
+
+CREATE TABLE creator (
+	creator_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,age_range ENUM
+	,Gender ENUM
+	,PRIMARY KEY (creator_id)
+	);
+
+CREATE TABLE smoothie (
+	smoothie_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,category VARCHAR(255)
+	,ingredients INT NOT NULL
+	,creator INT NOT NULL
+	,FOREIGN KEY (category) REFERENCES colorType(category)
+	,FOREIGN KEY (ingredients) REFERENCES ingredient(ingredient_id)
+	,FOREIGN KEY (creator) REFERENCES creator(creator_id)
+	,PRIMARY KEY (smoothie_id)
+	);
+
+CREATE TABLE type (
+	type_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,PRIMARY KEY (type_id)
+	,
+	);
+
+CREATE TABLE color (
+	color_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,PRIMARY KEY (color_id)
+	,
+	);
+
+CREATE TABLE ingredient (
+	ingredient_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,type_id INT NOT NULL
+	,color_id INT NOT NULL
+	,PRIMARY KEY (ingredient_id)
+	,FOREIGN KEY (type_id) REFERENCES type(type_id)
+	,FOREIGN KEY (color_id) REFERENCES color(color_id)
+	);
+
+CREATE TABLE colorType (
+	color_id INT NOT NULL
+	,type_id INT NOT NULL
+	,category VARCHAR(255)
+	,FOREIGN KEY (color_id) INT NOT NULL REFERENCES color(color_id)
+	,FOREIGN KEY (type_id) INT NOT NULL REFERENCES type(type_id)
+	,PRIMARY KEY (
```

Comment:
You used a combination of two foreign keys, which is possible to use, but it will work only if the combination of these two columns (type_id and color_is) is unique for each row in the table. Two rows can not have the same values for both columns.

Reply:
We were explained in the class that for the bridge table we use a composite primary key that consists of two foreign keys. This ensures that there cannot be any duplicates in the table.
Should I add some other keywords or commands on top of that?

Reply:
What I meant is does the combination of color and type are unique for each row? Because primary key should be unique. If so then you definitely can use it.